test/test_dists.jl

```diff
@@ -1,245 +1,332 @@
 # Unit tests for Distances
 
-@testset "individual metrics" begin
+macro test_metricity(_dist, _x, _y, _z)
+    quote
+        dist = $(esc(_dist))
+        x = $(esc(_x))
+        y = $(esc(_y))
+        z = $(esc(_z))
+        dxy = evaluate(dist, x, y)
+        dxz = evaluate(dist, x, z)
+        dyz = evaluate(dist, y, z)
+        if isa(dist, PreMetric)
+            # Unfortunately small non-zero numbers (~10^-16) are appearing
+            # in our tests due to accumulating floating point rounding errors.
+            # We either need to allow small errors in our tests or change the
+            # way we do accumulations...
+            @test evaluate(dist, x, x) + one(eltype(x)) ≈ one(eltype(x))
```

Comment:
No, it doesn't - again there are too many small errors in accumulation of the floating point numbers. The probabilities that these often represent are ≈ 1, so I thought that was a reasonable precision to use - otherwise I'd have been picking some arbitrary error tolerance using approx() that should more reasonably have depended on the floating point precision being used. You're right it could also be put into a function - in fact I had it as that earlier - but I didn't feel it added to the clarity of the code.